test/solvers/test_adaptive_regularization_with_cubics.jl

```diff
@@ -126,7 +122,14 @@ include("../utils/example_tasks.jl")
         @test isapprox(M, p1, p2)
         # Third with approximate Hessian
         p3 = adaptive_regularization_with_cubics(
-            M, f, grad_f, p0; θ=0.5, σ=100.0, retraction_method=PolarRetraction()
+            M,
+            f,
+            grad_f,
+            p0;
+            θ=0.5,
+            σ=100.0,
+            retraction_method=PolarRetraction(),
+            debug=[:Iteration, :Cost, :σ, :ρ_denonimator, "\n"],
```

Comment:
`ρ_denonimator` looks like a typo?

Reply:
Maybe. I am basically lost at this moment, because running the tests on Julia 1.6 locally on my Mac OS the tests pass, running them on CI '1.6 they fail. 
But you are right, the variable should be called nominator actually – still this is just a wrong naming of a variable, but not the reason for this to fail. Locally on 1.6 (on 1.9 anyways) tests pass, just on CI they don't.
And yes I spent about 3 hours with that yesterday, so I might be a bit annoyed by now.

Reply:
It says `ρ_denoNIMator`, not `ρ_denoMINator` :wink: . BTW, the opposite of denominator is numerator. 

Unfortunately I can't help much regarding that error on CI, I know almost nothing about ARC.

Reply:
Hehe, nice, even better. But that typo we then have for about a few months. Sure I can fix that once I fix the bug.

It seems a tolerance issue for some approximate hessians on exactly Mac OS 1.6 because the value that is not allowed to be negative gets to about `-1e-10`; with the new error system I could also go for just issuing a warning, maybe.